calico-cloud_versioned_docs/version-3.15/visibility/elastic/flow/datatypes.mdx

```diff
@@ -64,3 +64,38 @@ The following table details the key/value pairs in the JSON blob, including thei
 | `tcp_total_retransmissions`       | long              | Total retranmitted packets. This field is only available if flowLogsEnableTcpStats is enabled                                                                                                                                                                                                                                                        |
 | `tcp_lost_packets`                | long              | Total lost packets. This field is only available if flowLogsEnableTcpStats is enabled                                                                                                                                                                                                                                                                |
 | `tcp_unrecovered_to`              | long              | Total unrecovered timeouts. This field is only available if flowLogsEnableTcpStats is enabled                                                                                                                                                                                                                                                        |
+
+### Format of the policies field
+
+The `policies` field contains a comma-delimited list of policy rules that matched the flow.  Each entry in the
+list has the following format:
+
+```
+<index>|<tier name>|<policy name>|<action>|<rule index>
+```
+
+Where,
+
+* `<index>` numbers the order in which the rules were hit, starting with `0`.
+  :::tip
+  Sort the entries of the list by the `<index>` to see the order that rules were hit.
+  :::
+
+* `<tier name>` is the name of the policy tier containing the policy, or `__PROFILE__` for a rule derived from a
+  `Profile` resource (this is the internal datatype used to represent a Kubernetes namespace and its associated
+  "default allow" rule).
+* `<policy name>` is the name of the policy/profile; its format depends on the type of policy:
+
+  * `<tier>.<name>` for {{prodname}} `GlobalNetworkPolicy`.
+  * `<namespace>/<name>` for Kubernetes `NetworkPolicy`.
+  * `<namespace>/<tier>.<name>` for {{prodname}} `NetworkPolicy`.
+
+  Staged policy names are prefixed with "staged:".
+
+* `<action>` is the action performed by the rule; one of `allow`, `deny`, `pass`.
+* `<rule index>` if non-negative, is the index of the rule that was matched within the policy, starting with 0.
+  Otherwise, a special value:
+
+  * `-1` means that the flow didn't match any policy. It hit the default action for the tier.  In this case, the
+    policy name is an arbitrary policy that was active for the endpoint.
+  * `-2` means "unknown". The rule index was not recorded.
```

Comment:
Can this actually happen?  I believe it's possible to get gaps in the the list (e.g. there's a tier 0, 1 and 3 because the NFLOG for tier 2 was lost), but I can't think of a situation where we'd create the entry but not have a rule index?

Reply:
I'm finding it really difficult to tell.  It's a value we use internally, not sure if we an emit a log with ti.